### ISSUE TITLE
Inconsistent result depending on random/small variation in a Markdown file

### ISSUE THREAD

**Issue by ccoVeille** [jdkato/txtlint#847]
### Check for existing issues

- [X] Completed

### Environment

Linux
Homebrew
vale 3.4.2

### Describe the bug / provide steps to reproduce it

I'm testing a rule to detect GitHub case and suggest fixing it

Here is the rule

```yaml
---
extends: substitution
message: Use '%s' instead of '%s'
level: warning
ignorecase: true
swap:
  GitHub: GitHub
```

Here is a test file:

github_test2.md

```markdown
http://github.com bar

[github](#whatever)
[github1](#whatever)
[github.](#whatever)

https://jwodder.github.io/
http://github.com

whatever
```

Code reports
- github.com bar because he wants www.GitHub.com (not convenient but not related to my bug)
- `[github](#whatever)` as it should be  `[GitHub](#whatever)` :heavy_check_mark: 
- `[github.](#whatever)` as it should be  `[GitHub.](#whatever)` :heavy_check_mark: 

github_test3.md

```markdown
http://github.com bar

[github](#whatever)
[github1](#whatever)
[github.](#whatever)

https://jwodder.github.io/

http://github.com
```

The only difference is on last line whatever was replaced by http://github.com

Code reports differs:
- github.com bar is still reported :heavy_check_mark: 
- `[github](#whatever)` is reported as it should be  `[GitHub](#whatever)` :heavy_check_mark: 
- `[github.](#whatever)` is no longer reported :thinking: :bug: 
- `https://jwodder.github.io/` is reported as invalid :thinking: :bug: 

Now let's change slightly

```markdown
bar

[github](#whatever)
[github1](#whatever)
[github.](#whatever)

https://jwodder.github.io/

http://github.com
```

So the on first line github.com was removed

Code reports differs again and seems normal:
- `[github](#whatever)` is reported as it should be  `[GitHub](#whatever)` :heavy_check_mark: 
- `[github.](#whatever)` as it should be  `[GitHub.](#whatever)` :heavy_check_mark: 
- `https://jwodder.github.io/` is no longer reported as invalid :thinking: but it's what I would have expected

**Comment by ccoVeille** [comment]
Here are the files

[github_test2.md](https://github.com/user-attachments/files/15523466/github_test2.md)
[github_test3.md](https://github.com/user-attachments/files/15523465/github_test3.md)
[github_test4.md](https://github.com/user-attachments/files/15523464/github_test4.md)
[GitHub.yml](https://github.com/user-attachments/files/15523470/GitHub.yml.txt)
